src/ServiceControl/Connection/PlatformConnectionDetails.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace ServiceControl.Connection
+{
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+
+    public class PlatformConnectionDetails
+    {
+        readonly ConcurrentDictionary<string, object> values = new ConcurrentDictionary<string, object>();
+
+        public void Add(string key, object value)
+        {
+            if (values.TryAdd(key, value))
+            {
+                return;
+            }
+
+            // Add a numeric suffix to duplicated keys
+            var suffix = 0;
+            do
+            {
+                suffix++;
+            } while (!values.TryAdd($"{key}{suffix}", value));
```

Comment:
Should we do this only when the key values a different e.g. if there are two audit instances but have the same audit queue name we should not require any user intervention.

Reply:
The ones which are likely to be duplicates (Message and Saga Audit) are deserialized as a dynamic json object which makes it challenging to determine equality. I think this is an optimization that can come later.

If they aren't equal, then we wouldn't be able to deduplicate, we'll show both, and the customer will have to hand-edit the file to get their desired results.

If they are equal, they we could in theory deduplicate but if we don't then the endpoint will still get details that work.

Reply:
Makes sense.